pkg/compact/compact.go

```diff
@@ -470,6 +474,70 @@ func (cg *Group) Resolution() int64 {
 	return cg.resolution
 }
 
+// should return the results/metrics of the planning simulation
+type PlanSim interface {
+	Simulate(ctx context.Context, g *DefaultGrouper, p *largeTotalIndexSizeFilter, sy *Syncer) error
+}
+
+type DefaultPlanSim struct {
+	grouper *DefaultGrouper
+	planner *largeTotalIndexSizeFilter
+	sy      *Syncer
+}
+
+func NewDefaultPlanSim(grouper *DefaultGrouper, planner *largeTotalIndexSizeFilter, sy *Syncer) *DefaultPlanSim {
```

Comment:
Let's change`*largeTotalIndexSizeFilter` to `Planner`? The planner should be a interface here.

Reply:
Also let's make grouper to interface type `Grouper` as well